buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-Personal

```diff
@@ -0,0 +1,361 @@
+                }
+            }
+        } finally {
+            try{
+                cleanWs()
+            } catch(e) {
+                def writer = new StringWriter()
+                e.printStackTrace(new PrintWriter(writer))
+                echo e.toString()
+            }
+        }
+    }
+
+    // launch all pipeline builds
+    parallel builds
+}
+
+/*
+* Checks the build input arguments.
+*/
+def validate_arguments() {
+    params.each { key, value ->
+        if (key.contains('_REPO')) {
+            // check if HTTPS protocol is used for github.com (external) repositories
+            if (value.contains('git@github.com:')) {
+                 error("${key} - ${value} - must use HTTPS: e.g. " + value.replace('git@github.com:', 'https://github.com/', ) + "!")
+            }
+
+            //  check if SSH protocol is used for other repositories (assumed private repositories) 
+            if (value.startsWith('https://') && !value.contains('https://github.com/')) {
+                // assuming is private Git repo, should use SSH not HTTPS protocol
+                error("${key} - ${value} - must use SSH: e.g. " + value.replace('https://', 'git@').replace(':', '/') + "!")
+            }
+        }
+
+        if (key.equals("TESTS_TARGETS")) {
+            def TOKENS = value.trim().tokenize(",")
+            for (TOKEN in TOKENS) {
+                if (!['_sanity', '_extended', 'none'].contains(TOKEN.trim())) {
+                    error("Unknown test target: ${TOKEN}")
+                }
+            }
+        }
+    }
+}
+
+/*
+* Parses the platforms build parameter to identify the specs.
+* Returns an array of specs. 
+*/
+def get_specs(SUPPORTED_SPECS) {
+    def SPECS = []
+
+    if (!params.PLATFORMS) {
+        error("Must specify PLATFORMS!")
+    }
+
+    if (params.PLATFORMS.trim() == 'all') {
+        return SUPPORTED_SPECS
+    }
+
+    for (platform in PLATFORMS.split(",")) {
+        platform = platform.trim()
+        if (!SUPPORTED_SPECS.contains(platform)) {
+            error("Unknown or unsupported platform: ${platform}")
+        }
+
+        SPECS.add(platform)
+    }
+
+    return SPECS
+}
+
+/*
+* Parses the build parameters to identify the Java versions.
+* Returns an array of versions.
+*/
+def get_versions() {
+    def VERSIONS = []
+
+    params.each { key, value  ->
+        if ((key.indexOf('Java') != -1) && (value == true)) {
+            VERSIONS.add(key.substring(4))
+        }
+    }
+
+    if (VERSIONS.size() == 0) {
+        error("Must specify Java version, e.g. Java8")
+    }
+
+    return VERSIONS
+}
+
+/*
+* Returns the short version of a given SHA.
+*/
+def get_short_sha(SHA) {
+    return sh (script: "git rev-parse --short ${SHA}", returnStdout: true).trim()
+}
+
+/*
+* Returns the Jenkins build and test pipeline.
+*/
+def build(OPENJDK_REPO, OPENJDK_BRANCH, OPENJDK_SHA, OPENJ9_REPO, OPENJ9_BRANCH, OPENJ9_SHA, OMR_REPO, OMR_BRANCH, OMR_SHA, SPEC, SDK_VERSION, BUILD_NODE, TEST_NODE) {
```

Comment:
The reformat of the pipelines scripts is a work in progress.

Reply:
@AdamBrousseau  fyi - reformat some more:
- move and rename Pipeline-Build-Test-Personal.build() to pipeline-functions.build_pipeline() and change its signature
-  move all the logic that initialize variables to variables-functions
- update the pipeline-functions.get_shas() to fetch shas for all OpenJDK repositories (8, 9, 10) when OPENJDK_* arguments are instances of Map. No functionality changes if OPENJDK_* arguments are strings. This would works for all type of pipelines.